src/types.ts

```diff
@@ -755,7 +760,37 @@ export interface DraxScrollViewProps extends ScrollViewProps, DraxAutoScrollProp
 	id?: string;
 }
 
-/** Event data for when an item is moved within a DraxList, reordering the list */
+/** DraxList item being dragged */
+export interface DraxListDraggedItemData<TItem> {
+	/* The list index of the item that is being dragged */
+	index: number;
+	/* The item that is being dragged (or undefined if data is not found) */
+	item?: TItem;
+}
+
+/** Event data for when a list item reorder drag action begins */
+export interface DraxListOnItemDragStartEventData<TItem>
+	extends DraxDragEventData, DraxListDraggedItemData<TItem> {}
+
+/** Event data for when a list item position (index) changes during a reorder drag */
+export interface DraxListOnItemDragPositionChangeEventData<TItem>
+	extends DraxMonitorEventData, DraxListDraggedItemData<TItem> {
+	/* The list index of the item position dragged over, or undefined if none */
+	toIndex: number | undefined;
+	/* The previous list index of the item position dragged over, or undefined if none */
+	previousIndex: number | undefined;
+}
```

Comment:
One thing I wanted to bring attention to was the way I changed `onItemDragPositionChange`. The event receives the current position as `toIndex` as well as the previous position `previousIndex` since we have it and it might be handy. You'll notice though that I have these as `number | undefined`. With this version of the logic, when you drag an item over another item's position, you get that item's index. When you drag it anywhere else - i.e., its own position, the list header/footer, outside the list - you get `undefined`. I thought it might be nice to return its own index when the item is hovering over its original position, but that is not easy for me to distinguish from those other cases (particularly the list header/footer).

The reason for this is the method used to identify which position a list item is being dragged over. Each of the items in the list is a DraxView which can be dragged and can receive. When we drag an item, the other list items are receiving it, and the monitor is seeing that. An item can't receive itself, so there is no event triggered to say when the item is dragged back over its own position. We could possibly enhance the list monitoring logic with a special case to check the in-list boundaries of the dragged item and distinguish "dragged over item's original position" from "dragged over a part of the list that has no item". Then we would be able to provide the item's own index as `toIndex` in this event, rather than `undefined`, when an item is dragged over its original position. I am not sure how crucial that distinction is in practice, so I'm open to feedback from actual use cases.

Reply:
@special-character if you think it would be important to have that functionality now, I could try to work it into this PR. Otherwise I might create a separate enhancement issue in the backlog.

Reply:
I spun this off into #45 and will merge the current PR as is.